src/components/Table/TableBasic.part.tsx

```diff
@@ -154,17 +154,24 @@ export class TableBasic<T> extends React.Component<TableProps<T> & RefProps, Tab
     } else if (this.isSortable(key, getColumns(data, columns))) {
       const { sorting } = this.state;
       const isAscending = sorting && sorting.order === 'descending' && sorting.columnKey === key;
+      const order = sorting && sorting.columnKey === key ? 'descending' : 'ascending';
+      let newSortingValue: TableSorting | undefined = undefined;
 
       if (!isAscending && column !== -1) {
-        this.setState({
-          sorting: {
-            columnKey: key,
-            order: sorting && sorting.columnKey === key ? 'descending' : 'ascending',
-          },
-        });
-      } else {
-        this.setState({
-          sorting: undefined,
+        newSortingValue = {
+          columnKey: key,
+          order,
+        };
+      }
+
+      this.setState({ sorting: newSortingValue });
```

Comment:
I suggest triggering `onSort` function inside `callback` argument of `setState`:
```ts
this.setState({ sorting: newSortingValue }, () => {
  if (typeof onSort === 'function') {
    onSort({
      column,
      key,
      value: newSortingValue && newSortingValue.order,
      row: -1,
    });
  }
});
```

Reply:
Same here https://github.com/ZEISS/precise-ui/pull/187/files#diff-0313273806b6ac65b35d01494c63ab75R214

Reply:
Updated it accordingly